### PR TITLE
WIP: [AIRFLOW-4506] Make GIT_SYNC_DEPTH and GIT_SYNC_REV configurable

### PR DESCRIPTION
**Pull request by andrejbaran**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by andrejbaran** [comment]
Hey @dimberman I'm not sure whether this is the correct approach to default values in this case as I want to keep it backward compatible (meaning no entries in `airflow.cfg`)

**Comment by dimberman** [comment]
Hi @andrejbaran I'm not sure what the question is? This seems in-line with existing stuff. That said, could you add a test for this please?

File: airflow/kubernetes/worker_configuration.py [hunk 1]
```diff
@@ -64,10 +64,13 @@ def _get_init_containers(self):
             'value': self.kube_config.git_sync_dest
         }, {
             'name': 'GIT_SYNC_DEPTH',
-            'value': '1'
+            'value': self.kube_config.git_sync_depth if self.kube_config.git_sync_depth else '1'
```
Comment (ashb) [inline airflow/kubernetes/worker_configuration.py:67]: Don't' put the default in code, just put the default in the config file.

**Review by ashb — CHANGES_REQUESTED** [review]
Move defaults for both new config entries out of the code into the config file.

**Comment by dimberman** [comment]
@andrejbaran any word on this? Would love to get it merged ASAP :).

**Comment by mik-laj** [comment]
@andrejbaran  Can I help you with this change?

**Comment by andrejbaran** [comment]
@mik-laj @dimberman hey guys, sorry for not responding. I will try to finish this up next week.

**Comment by KevinKobi** [comment]
`GIT_SYNC_REV` was added https://github.com/apache/airflow/pull/6350 but not `GIT_SYNC_DEPTH` :( 
are there plans to add it?

**Pull request closed without merging**